ruby_basics/bubble_sort/bubble_sort.rb

```diff
@@ -0,0 +1,56 @@
+# assignment: https://www.theodinproject.com/lessons/ruby-bubble-sort#assignment
+
+# answer: 
+# > bubble_sort([4,3,78,2,0,2])
+# => [0,2,2,3,4,78]
+
+# everything just keeps getting more difficult... 
+
+def bubble_sort(array)  
+  swapping = true 
+
+  while swapping 
+    swapping = false
+
+    array.each_with_index do |number, index|
+
+      # if the number or the next number are nil (end of array)
+      # immediately stop the code
+      if number.nil? || array[index+1].nil?
```

Comment:
I don't think you need to check `number.nil?`, as I don't think that can ever happen. 

I removed it and this program still works.